app/src/main/java/io/appium/settings/Settings.java

```diff
@@ -79,26 +62,17 @@ public void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.main);
         Log.d(TAG, "Entering the app");
 
-        registerSettingsReceivers(Arrays.asList(
-                WiFiConnectionSettingReceiver.class,
-                AnimationSettingReceiver.class,
-                DataConnectionSettingReceiver.class,
-                LocaleSettingReceiver.class,
-                LocalesReader.class,
-                LocationInfoReceiver.class,
-                ClipboardReceiver.class,
-                BluetoothConnectionSettingReceiver.class,
-                UnpairBluetoothDevicesReceiver.class,
-                NotificationsReceiver.class,
-                SmsReader.class,
-                MediaScannerReceiver.class
-        ));
-
-        // https://developer.android.com/about/versions/oreo/background-location-limits
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            startForegroundService(ForegroundService.getForegroundServiceIntent(Settings.this));
+        String recordingAction = getIntent().getAction();
+        if (recordingAction != null && recordingAction.startsWith(ACTION_RECORDING_BASE)) {
+            Log.d(TAG, "Skip starting foreground service");
         } else {
-            LocationTracker.getInstance().start(this);
+            // https://developer.android.com/about/versions/oreo/background-location-limits
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                startForegroundService(ForegroundService.getForegroundServiceIntent(Settings.this));
+            } else {
+                SettingsReceivers.register(getApplicationContext());
+                LocationTracker.getInstance().start(this);
+            }
```

Comment:
where do we want to unregister services in OSs older than Android O?

Reply:
Good point. I was wondering if we needed to add `Application` lifecycle, but I haven't investigated it yet.

I'd have liked to try it out the method after this PR. If it works, maybe the service's onDestroy also can move to the same place to "unregister"

Reply:
Without that, currently we have no proper timing to "unregister" since it is not Activity's lifecycle